lib/core/connector_settings.rb

```diff
@@ -80,20 +83,33 @@ def scheduling_settings
       self[:scheduling]
     end
 
+    def filtering
+      Utility::Commons.return_if_present(@elasticsearch_response[:filtering], DEFAULT_FILTERING)
+    end
+
+    def active_filter_config
+      # assume for now, that first object in filtering array or a filter object itself is the only filtering object
+      filter = filtering.is_a?(Array) ? filtering[0] : filtering
```

Comment:
Might be a really good idea to warn/raise error if unexpected values that we can transparently handle happens